tests/reorder_python_imports_test.py

```diff
@@ -586,49 +585,143 @@ def test_replace_module_imported_asname():
+    ),
+    pytest.param(
+        "# I'm a license comment\n"
+        'import os\n',
 
+        "# I'm a license comment\n"
+        'import os\n',
 
-@tfiles
-def test_integration_main(filename, tmpdir):
-    with io.open(os.path.join('test_data/inputs', filename)) as f:
-        input_contents = f.read()
-    with io.open(os.path.join('test_data/outputs', filename)) as f:
-        expected = f.read()
+        id='license comment',
+    ),
+    pytest.param(
+        'import reorder_python_imports\n'
+        'import os\n'
+        'import six\n',
+
+        'import os\n\n'
+        'import six\n\n'
+        'import reorder_python_imports\n',
 
+        id='needs reordering',
+    ),
+    pytest.param(
+        'import sys\n'
+        'import os',
+
+        'import os\n'
+        'import sys\n',
+
+        id='no eol',
+    ),
+    pytest.param(
+        '# noreorder\n'
+        'import reorder_python_imports\n'
+        'import os\n'
+        'import six\n',
+
+        '# noreorder\n'
+        'import reorder_python_imports\n'
+        'import os\n'
+        'import six\n',
+
+        id='noreorder all',
+    ),
+    pytest.param(
+        'import sys\n'
+        'import reorder_python_imports\n\n'
+        'import matplotlib # noreorder\n'
+        "matplotlib.use('Agg')\n",
+
+        'import sys\n\n'
+        'import reorder_python_imports\n\n'
+        'import matplotlib # noreorder\n'
+        "matplotlib.use('Agg')\n",
+
+        id='noreorder inline',
+    ),
+    pytest.param(
+        'import sys\n'
+        'import reorder_python_imports\n\n'
+        '# noreorder\n'
+        'import matplotlib\n'
+        "matplotlib.use('Agg')\n",
+
+        'import sys\n\n'
+        'import reorder_python_imports\n\n'
+        '# noreorder\n'
+        'import matplotlib\n'
+        "matplotlib.use('Agg')\n",
+
+        id='noreorder not at beginning',
+    ),
+    pytest.param(
+        '',
+
+        '',
+
+        id='trivial',
+    ),
+)
+
+cases = pytest.mark.parametrize(('s', 'expected'), CASES)
+
+
+@cases
+def test_fix_file_contents(s, expected):
+    assert fix_file_contents(s) == expected
+
+
+@cases
+def test_integration_main(s, expected, tmpdir):
     test_file = tmpdir.join('test.py')
-    test_file.write(input_contents)
+    test_file.write(s)
 
     # Check return value with --diff-only
     retv_diff = main((str(test_file), '--diff-only'))
-    assert retv_diff == int(input_contents != expected)
+    assert retv_diff == int(s != expected)
 
     retv = main((str(test_file),))
     # Check return value
-    assert retv == int(input_contents != expected)
+    assert retv == int(s != expected)
 
     # Check the contents rewritten
     assert test_file.read() == expected
 
 
-def test_integration_main_stdout(capsys):
-    ret = main(('--print-only', 'test_data/inputs/needs_reordering.py'))
+def test_integration_main_stdout(tmpdir, capsys):
+    case = [c for c in CASES if c.id == 'needs reordering'][0]
```

Comment:
I would just inline a small example instead of trying hard to reuse the parametrize list